### ISSUE TITLE
ClosureCompiler vs unicode characters?

### ISSUE THREAD

**Issue by phloe** [Myndex/colorparsley#3]
Hi,

I'm having trouble using colorParsley due to an error that looks like a &copy; character in the code not being handled properly by ClosureCompiler?

<img width="730" alt="Screenshot 2022-05-15 at 18 28 31" src="https://user-images.githubusercontent.com/13237/168483329-1ee79eca-3d27-464f-a3df-2878af9d68f7.png">

The issue disappears as soon as the unicode character is removed.

Of course you could just remove the &copy; - or replace it with a simpler `(c)`...

BUT -  you could also avoid using ClosureCompiler altogether 🙂
Distributing minified code in npm packages is not really best practice; it's better to just distribute the raw source (if it's valid ESM/CommonJS).

The package is already marked as `"type": "module"` in the package.json  so all you need is the proper exports:

```js
// just add export in front of the function you want to expose
export function colorParsley (colorIn) { /* ... */ };

export function colorToHex (rgba = [0,0,0,''],allow3 = true) { /* ... */ }

export function colorToRGB (rgba = [0,0,0,'']) { /* ... */ }

// module.exports = { colorParsley, colorToHex, colorToRGB };
// ^ remove this
```

**Comment by Myndex** [comment]
Hi Rasmus @phloe 

THANK YOU for the feedback!!

>  _I'm having trouble using colorParsley due to an error that looks like a © character in the code not being handled properly by ClosureCompiler?_

Doh! I thought I had removed the unicode goofs... Will fix asap

-----
> _BUT - you could also avoid using ClosureCompiler altogether 🙂 Distributing minified code in npm packages is not really best practice; it's better to just distribute the raw source_ 

Really? _Monkey see monkey do_ for me I suppose, I noticed a lot of packages are that way, so I followed suit.

In truth, Color Parsley is a pretty small library, so minifying is probably one of those "needless optimizations" that are all the rage... LOL

**_QUESTION:_** Will there be any potential breakage if I put the source in the ` dist ` folder, and without the `.min.` in the name? As this is pre-release beta, such breakage is to be expected, but trying to understand these various patterns in the node.js ecosystem, etc...

-----
> _The package is already marked as "type": "module" in the package.json so all you need is the proper exports: `// just add export in front of the function you want to expose`_

In the file you'll see I actually have the export for each function commented out, as in:

```js
    //export
    function colorParsley (colorIn) { /* ... */ };
```
And have been using the 

```js
    module.exports = { colorParsley, colorToHex, colorToRGB };
```

Instead, partly on advice from another coder, and partly as this allows the file to work in my local development environment, which at present is not setup to handle working with the "server" configuration for things like exports...

I suppose I should setup my dev environment differently — right now, all the JS runs in the browser as client-side.

I just recently started publishing to npm (this and APCA are the first npm projects I've done) ... I'm used to doing things "in house" so I expect a stumble here or there in getting things all sorted for openSource and the node.js ecosystem etc.

That said, is there a doc or tutorial you can suggest for setting up my dev environment so it matches how people are most likely to use these packages in the wild? I have just been using GitHub desktop with BBEdit, and running the scripts in BBEdit or a browser. But it appears that does not support how people want or expect the packages to be built.

THANK YOU much for your comments and help. I'll also review the pull request too for the tests soon as I can.

**Comment by phloe** [comment]
If you look at the [test.html](https://github.com/Myndex/colorparsley/pull/4/files#diff-8b6b8086edd980b7fbc7b614359d89dd6dff96a48c9faf27e184bcd760892c3f) changes I made in the related PR - you'll notice that "local"/client and "server" is now just one syntax via `type="module"` and `import { x } from "x";` (ESM format). No reason to distinguish between the two anymore 🙂 
 
Most people probably use a terminal and a bundler (webpack, parcel, vite, snowpack, etc) when consuming npm packages - but writing small packages like this one doesn't require those at all. Github desktop and BBEdit should do just fine.

Of course - if you want to run tests you'll need a terminal (BBEdit has one built-in?) to run an `npm test` command and see it's output.

**Comment by Myndex** [comment]
Published (0.1.6) I want to thank you again for your help here!
